example/models/food/v1/food_test.go

```diff
@@ -0,0 +1,315 @@
+			require.NoErrorf(t, err, "TestSchemaFoodPasta: failed to execute sql: '\n%s\n'", line)
+			// os.Stderr.WriteString(line)
+			// os.Stderr.WriteString("\n------\n")
+		}
+		ct := schema[0]
+		require.Contains(t, ct, "CREATE TABLE")
+		require.Equal(t, 2,
+			strings.Count(ct, "$pksk"),
+			"Create table should contain only one pksk field + index: %s", ct,
+		)
+
+		if smsg.DBReflect().Descriptor().IsPartitioned() {
+			require.Contains(t, ct, "PARTITION BY LIST")
+		} else {
+			require.NotContains(t, ct, "PARTITION BY LIST")
+		}
+
+		require.Equal(t, 1,
+			strings.Count(ct, "fts_data"),
+			"Create table should contain only one fts_data field: %s", ct,
+		)
+
+		_, err = pg.DB.Exec(ctx, `ALTER TABLE `+smsg.DBReflect().Descriptor().TableName()+` DROP COLUMN "pb$id"`)
+		require.NoErrorf(t, err, "TestSchemaFoodPasta: failed to drop col id")
+
+		schema, err = pgdb_v1.MigratePartitions(ctx, pg.DB, smsg)
+		require.NoError(t, err)
+		for _, line := range schema {
+			//fmt.Println(line)
+			_, err := pg.DB.Exec(ctx, line)
+			require.NoErrorf(t, err, "TestSchemaFoodPasta: failed to execute sql: '\n%s\n'", line)
+			// os.Stderr.WriteString(line)
+			// os.Stderr.WriteString("\n------\n")
+		}
+		if len(schema) > 0 {
+			ct = schema[0]
+			require.Contains(t, ct, "ALTER TABLE")
+		}
+
+		fakeTenantIds := []string{"t1", "t2", "t3"}
+		protoTableName := smsg.DBReflect().Descriptor().TableName()
+
+		if smsg.DBReflect().Descriptor().IsPartitioned() {
+			// TODO(scott) makes helper functions and move to pgdb_v1
+			verifyMasterPartition(t, pg, protoTableName, fakeTenantIds)
+			// Test sub-tables for partitions
+			// Create sub-tables
+			for _, tenantId := range fakeTenantIds {
+				_, err = pg.DB.Exec(ctx, fmt.Sprintf("CREATE TABLE %s_%s PARTITION OF %s FOR VALUES IN ('%s')", protoTableName, tenantId, protoTableName, tenantId))
+				require.NoError(t, err, "TestSchemaFoodPasta: failed to create partitioned table")
+			}
+			verifySubTables(t, pg, protoTableName, fakeTenantIds)
+			// Insert data into master table
+			testInsertAndVerify(t, pg, protoTableName, fakeTenantIds, smsg)
+			testPartitionIndexing(t, pg, smsg)
+			verifyPartitionIndexes(t, pg, smsg, fakeTenantIds)
+
+		}
+	}
+
+}
+
+func testPartitionIndexing(t *testing.T, pg *pgtest.PG, smsg pgdb_v1.DBReflectMessage) {
+	ctx := context.Background()
+	// Update indexes on sub tables
+	schema, err := pgdb_v1.IndexPartitions(ctx, pg.DB, smsg)
+	require.NoError(t, err)
+	for _, line := range schema {
+		fmt.Printf("%s \n", line)
+		fmt.Println("END INDEXING")
+		_, err := pg.DB.Exec(ctx, line)
+		require.NoErrorf(t, err, "TestSchemaFoodPasta: failed to execute sql: '\n%s\n'", line)
+	}
+}
+
+// TODO(scott) this is correct but whats currently failing is that the indexes need to be unique per
+// partitioned table.  So we need to add the tenantId or some hash to the index name.
+func verifyPartitionIndexes(t *testing.T, pg *pgtest.PG, smsg pgdb_v1.DBReflectMessage, fakeTenantIds []string) {
```

Comment:
TODO
So something I found is that we need to add another hash for sub-tables so each of their indexes are unique

in tables I dont have any PG indexes its happy but when I add some they only get added to the first table since they already exist once the 2nd and 3rd table get indexed

```
FOR TENANT t1
pb_pasta_ingredient_models_food_v1_0565c036_t1_pkey
pb_pasta_ingredient_models_food_v1_pb$tenant_id_pb$pk_pb$sk_idx
pb_pasta_ingredient_models_food_v1_pb$tenant_id_pb$fts_data_idx

// Add another hash for these
pbidx_pastas_pasta_ingredient_models_food_v1_e3bd47ef
pbidx_ingredients_pasta_ingredient_models_foo_cc57095e
pbidx_everythingggg_pasta_ingredient_models_f_30c0b1c3

Index rows: 6
FOR TENANT t2
pb_pasta_ingredient_models_food_v1_0565c036_t2_pkey
pb_pasta_ingredient_models_food_v_pb$tenant_id_pb$pk_pb$sk_idx1
pb_pasta_ingredient_models_food_v_pb$tenant_id_pb$fts_data_idx1

Index rows: 3
```

Reply:
I dont want these to get extremely long either so maybe just a hash on the tenantId value?
```
pbidx_pastas_pasta_ingredient_models_food_v1_e3bd47ef_<tenantIdHash>
```

Reply:
Edit: or even just the tenantId itself would work

Reply:
https://github.com/ductone/protoc-gen-pgdb/pull/63/commits/2e2550bfa23a80c7e07d548ec041b01323208fb7